core/src/Kokkos_Parallel.hpp

```diff
@@ -352,10 +352,10 @@ inline void parallel_scan(const std::string& str, const ExecutionPolicy& policy,
   ExecutionPolicy inner_policy = policy;
   Kokkos::Tools::Impl::begin_parallel_scan(inner_policy, functor, str, kpID);
 
-  Kokkos::Impl::shared_allocation_tracking_disable();
-  Impl::ParallelScan<FunctorType, ExecutionPolicy> closure(functor,
-                                                           inner_policy);
-  Kokkos::Impl::shared_allocation_tracking_enable();
+  auto closure = Kokkos::Impl::with_shared_allocation_tracking_disabled([&]() {
+    return Impl::ParallelScan<FunctorType, ExecutionPolicy>(functor,
+                                                            inner_policy);
+  });
 
   closure.execute();
```

Comment:
Why don't we wait with reenabling allocation tracking until after the kernel is enqueued anyway?

Reply:
We don't need to -- any views in the functor get a sticky `DO_NOT_DEREF_FLAG` set, so reference counting is permanently disabled. So we don't have to worry about enabling/disabling reference counting anymore.